lit_gpt/lora.py

```diff
@@ -345,7 +339,7 @@ def merge(self) -> None:
                 0
             )  # (1, 4, 128) @ (256, 2, 1) -> (1, 256, 128) -> (256, 128)
             # W = W + delta_W (merge)
-            self.linear.weight.data += self.zero_pad(delta_w * self.scaling)  # (256, 128) after zero_pad (384, 128)
+            self.linear.weight.data += self.zero_pad(delta_w.T * self.scaling).T  # (256, 128) after zero_pad (384, 128)
```

Comment:
I think that it's better to do double transpose here (and once), rather than every time in `zero_pad` and for the cases where it's not needed at all.